cards/src/main/java/com/decathlon/vitamin/compose/cards/VitaminCards.kt

```diff
@@ -0,0 +1,258 @@
+        imageOverlayAlignment: Alignment = Alignment.TopStart,
+        imageOverlay: (@Composable () -> Unit)? = null,
+    ) = Card(modifier = modifier, shape = shape) {
+        Column {
+            CardImage(
+                painter = painter,
+                contentDescription = contentDescription,
+                modifier = Modifier.padding(if (imageScale == VitaminCardImageScale.Fit) 16.dp else 0.dp),
+                overlay = imageOverlay,
+                imageOverlayAlignment = imageOverlayAlignment
+            )
+            Box(
+                modifier = Modifier.padding(16.dp),
+            ) {
+                TopImageContent.content()
+            }
+        }
+    }
+
+    /**
+     * A card with full image in background
+     * @param painter the [Painter] resource for the image
+     * @param contentDescription to describe image for screen reader. Set to null only if image is decorative.
+     * @param content The content of the card. Can be a composable from [FullImageContent] or a custom composable.
+     * @param modifier The `Modifier` to be applied to the all card.
+     * @param shape to change the shape of the card.
+     * @param imageWithGradient A boolean value to add a gradient effect to the image. No gradient by default.
+     */
+    @Composable
+    fun FullImage(
+        painter: Painter,
+        contentDescription: String?,
+        content: @Composable (FullImageContent.() -> Unit),
+        modifier: Modifier = Modifier,
+        shape: Shape = RoundedCornerShape(4.dp),
+        imageWithGradient: Boolean = false,
+    ) = Card(modifier = modifier, shape = shape) {
+        Box {
+            CardImage(
+                painter = painter,
+                contentDescription = contentDescription,
+                withGradient = imageWithGradient
+            )
+            Box(modifier = Modifier.align(Alignment.BottomStart).padding(16.dp)) {
+                FullImageContent.content()
+            }
+        }
+    }
+
+    /**
+     * A card with thumbnail image at left
+     * @param painter the [Painter] resource for the image
+     * @param contentDescription to describe image for screen reader. Set to null only if image is decorative.
+     * @param content The content of the card. Can be a composable from [SideImageContent] or a custom composable.
+     * @param modifier The `Modifier` to be applied to the all card.
+     * @param actionContent to add actions icons at the top end of the card like clickable icon.
+     * Can be a composable from [SideImageActionContent] or a custom composable.
+     * @param shape to change the shape of the card.
+     */
+    @Composable
+    fun SideImage(
+        painter: Painter,
+        contentDescription: String?,
+        content: (@Composable SideImageContent.() -> Unit),
+        modifier: Modifier = Modifier,
+        actionContent: @Composable (SideImageActionContent.() -> Unit)? = null,
```

Comment:
I think we can have something more smart here. Because the space for the action work like the topbar, maybe we can have a similar mechanism? Even an overflow management with a dropdown?